### PR TITLE
Add TLS Termination support for FirewallPolicy

### PR DESCRIPTION
**Pull request by odedlaz**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2020-04-01/firewallPolicy.json [hunk 1]
```diff
@@ -622,10 +626,84 @@
...
+        }
+      },
+      "description": "Trusted Root certificates properties for tls."
+    },
+    "FirewallPolicyCertificateAuthority": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/FirewallPolicyCertificateAuthorityPropertiesFormat",
+          "description": "Properties of the certificate authority"
```
Comment (number213) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-04-01/firewallPolicy.json:689]:
```suggestion
          "description": "Properties of the certificate authority."
```

**Review by anton-evseev — CHANGES_REQUESTED** [review]
LGTM except one nitpick

File: specification/network/resource-manager/Microsoft.Network/stable/2020-04-01/firewallPolicy.json (hunk 1 above)
Comment (odedlaz) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-04-01/firewallPolicy.json]: I forced-pushed a fix.

**Review by anton-evseev — APPROVED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2020-04-01/firewallPolicy.json [hunk 2]
```diff
@@ -622,10 +626,84 @@
...
+        "transportSecurity": {
+          "description": "TLS Configuration definition.",
+          "$ref": "#/definitions/FirewallPolicyTransportSecurity"
         }
       },
       "description": "Firewall Policy definition."
     },
+    "FirewallPolicyTransportSecurity": {
+      "properties": {
+        "certificatesAuthority": {
```
Comment (majastrz) [inline specification/network/resource-manager/Microsoft.Network/stable/2020-04-01/firewallPolicy.json:639]: >s [](start = 20, length = 1)

This looks like a typo and should be `certificateAuthority`

Comment (odedlaz) [reply specification/network/resource-manager/Microsoft.Network/stable/2020-04-01/firewallPolicy.json]: @majastrz, @number213 , done.

**Review by majastrz — APPROVED** [review]
Signed off from ARM side.

**Pull request merged**